gtsam/slam/dataset.cpp

```diff
@@ -585,8 +586,14 @@ BetweenFactorPose3s parse3DFactors(const string& filename) {
       mgtsam.block<3, 3>(3, 0) = m.block<3, 3>(3, 0);  // off diagonal
 
       SharedNoiseModel model = noiseModel::Gaussian::Information(mgtsam);
+      auto R12 = Rot3::Quaternion(qw, qx, qy, qz);
+      if (corruptingNoise) {
+        Sampler sampler(corruptingNoise);
```

Comment:
No, please move it out of the loop at least?? You can make an optional Sampler, and only sample if it evaluates to True here